go/tasks/plugins/k8s/pytorch/pytorch_test.go

```diff
@@ -0,0 +1,354 @@
+)
+
+const testImage = "image://"
+const serviceAccount = "pytorch_sa"
+
+var (
+	dummyEnvVars = []*core.KeyValuePair{
+		{Key: "Env_Var", Value: "Env_Val"},
+	}
+
+	testArgs = []string{
+		"test-args",
+	}
+
+	resourceRequirements = &corev1.ResourceRequirements{
+		Limits:  corev1.ResourceList{
+			corev1.ResourceCPU: 		resource.MustParse("1000m"),
+			corev1.ResourceMemory:		resource.MustParse("1Gi"),
+			flytek8s.ResourceNvidiaGPU:	resource.MustParse("1"),
+		},
+		Requests: corev1.ResourceList{
+			corev1.ResourceCPU: 		resource.MustParse("100m"),
+			corev1.ResourceMemory:		resource.MustParse("512Mi"),
+			flytek8s.ResourceNvidiaGPU:	resource.MustParse("1"),
+		},
+	}
+
+	jobName = "the-job"
+	jobNamespace = "pytorch-namespace"
+)
+
+func dummyPytorchCustomObj(workers int32) *plugins.PyTorchOperatorTask {
+	return  &plugins.PyTorchOperatorTask{
+		Workers:              workers,
+	}
+}
+
+func dummySparkTaskTemplate(id string, pytorchCustomObj *plugins.PyTorchOperatorTask) *core.TaskTemplate {
+
+	ptObjJSON, err := utils.MarshalToString(pytorchCustomObj)
+	if err != nil {
+		panic(err)
+	}
+
+	structObj := structpb.Struct{}
+
+	err = jsonpb.UnmarshalString(ptObjJSON, &structObj)
+	if err != nil {
+		panic(err)
+	}
+
+	return &core.TaskTemplate{
+		Id:   &core.Identifier{Name: id},
+		Type: "container",
+		Target: &core.TaskTemplate_Container{
+			Container: &core.Container{
+				Image: testImage,
+				Args:  testArgs,
+				Env:   dummyEnvVars,
+			},
+		},
+		Custom: &structObj,
+	}
+}
+
+
+func dummyPytorchTaskContext(taskTemplate *core.TaskTemplate) pluginsCore.TaskExecutionContext {
+	taskCtx := &mocks.TaskExecutionContext{}
+	inputReader := &pluginIOMocks.InputReader{}
+	inputReader.OnGetInputPrefixPath().Return(storage.DataReference("/input/prefix"))
+	inputReader.OnGetInputPath().Return(storage.DataReference("/input"))
+	inputReader.OnGetMatch(mock.Anything).Return(&core.LiteralMap{}, nil)
+	taskCtx.OnInputReader().Return(inputReader)
+
+	outputReader := &pluginIOMocks.OutputWriter{}
+	outputReader.On("GetOutputPath").Return(storage.DataReference("/data/outputs.pb"))
+	outputReader.On("GetOutputPrefixPath").Return(storage.DataReference("/data/"))
+	taskCtx.On("OutputWriter").Return(outputReader)
+
+	taskReader := &mocks.TaskReader{}
+	taskReader.On("Read", mock.Anything).Return(taskTemplate, nil)
+	taskCtx.On("TaskReader").Return(taskReader)
+
+	tID := &mocks.TaskExecutionID{}
+	tID.On("GetID").Return(core.TaskExecutionIdentifier{
+		NodeExecutionId: &core.NodeExecutionIdentifier{
+			ExecutionId: &core.WorkflowExecutionIdentifier{
+				Name:    "my_name",
+				Project: "my_project",
+				Domain:  "my_domain",
+			},
+		},
+	})
+	tID.On("GetGeneratedName").Return("some-acceptable-name")
+
+
+	resources := &mocks.TaskOverrides{}
+	resources.OnGetResources().Return(resourceRequirements)
+
+	taskExecutionMetadata := &mocks.TaskExecutionMetadata{}
+	taskExecutionMetadata.On("GetTaskExecutionID").Return(tID)
```

Comment:
we are trying to migrate everything to use
OnGetTraskExecutionID instead of On("GetTaskExecutionID") -> this helps in compile time safety. Can you please change all the "quoted" method references